python3Packages.rich: init at 8.0.0

sengaya opened the pull request:
###### Motivation for this change

Dependency for latest version of ansible-lint

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

I could run the unit tests when using `fetchFromGitHub` instead of `fetchPypi` but then `pythonImportsCheck` failed. I sticked to `fetchPypi` for now and disabled the unit tests.

sengaya committed:
python3Packages.rich: init at 8.0.0

mweinelt commented on pkgs/development/python-modules/rich/default.nix:
```diff
@@ -0,0 +1,24 @@
+
+  propagatedBuildInputs = [ typing-extensions pygments recommonmark colorama ];
+
+  doCheck = false;
```
If the pypi package does not contain the tests please switch to the GitHub sources.

sengaya replied:
Switched to GitHub sources. I had to patch `setup.py` though, see 9651c26b434f047cc4b7d3764ae0591ceee2b902.

mweinelt commented on pkgs/development/python-modules/rich/default.nix:
```diff
@@ -0,0 +1,24 @@
+buildPythonPackage rec {
+  pname = "rich";
+  version = "8.0.0";
+  disabled = isPy27;
```
The difference is probably minimal, but `pythonOlder "3.6"` is probably more accurate? https://github.com/willmcgugan/rich/blob/master/pyproject.toml#L26

sengaya committed:
Switch to fetchFromGitHub, use pythonOlder

- Use fetchFromGitHub, needed for enabling tests
- Use pythonOlder to match exact requirement of the package

mweinelt commented on pkgs/development/python-modules/rich/default.nix:
```diff
@@ -0,0 +1,31 @@
+  };
+
+  postPatch = ''
+   sed -i 's/setuptools.setup(name="rich")/setuptools.setup(name="rich", packages=setuptools.find_packages())/g' setup.py
```
indent is one space off

mweinelt commented on pkgs/development/python-modules/rich/default.nix:
```diff
@@ -0,0 +1,31 @@
+
+  propagatedBuildInputs = [ typing-extensions pygments recommonmark colorama ];
+
+  checkInputs = [ pytest pytestcov ];
```
We generally don't require coverage tests, since they are of no use to us. If you can skip them easily that would be appreciated.

jonringer replied:
seeing as the test suite is invoked through "make test", it may be required there.

However, it would probably be best to look at the command that `make test` is running, and just run it without coverage plugins.

sengaya replied:
Thanks for the feedback. This is the command called: [`pytest --cov-report term-missing --cov=rich tests/ -vv`](https://github.com/willmcgugan/rich/blob/master/Makefile#L2). I guess the `-vv` can be omitted as well? I'll update the PR.

sengaya committed:
Remove coverage tests, formatting

sengaya commented:
I guess that was bad timing and we can close this PR: 5b66067eb9fef083f06856e8145f471ad8d37c4e
